controllers/apps/apimanager_controller.go

```diff
@@ -205,17 +205,22 @@ func (r *APIManagerReconciler) apiManagerInstance(namespacedName types.Namespace
 }
 
 func (r *APIManagerReconciler) setAPIManagerDefaults(cr *appsv1alpha1.APIManager) (reconcile.Result, error) {
-	externalChanged := cr.HighAvailabilityToExternalComponents()
-	changed, err := cr.SetDefaults() // TODO check where to put this
+	updated := false
+
+	externalChanged := cr.UpdateExternalComponentsFromHighAvailability()
+	updated = updated || externalChanged
```

Comment:
```suggestion
	updated := cr.UpdateExternalComponentsFromHighAvailability()
```

Just a tiny change. Since `updated` is set to `false` before being `or`ed with `externalChanged` it's pretty much redundant, we can assign it directly from the return value of `UpdateExternalComponentsFromHighAvailability`